spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandler.java

```diff
@@ -357,6 +358,22 @@ protected void onInit() {
 		this.forceReleaseProcessor = createGroupTimeoutProcessor();
 	}
 
+	@Override
+	public void afterSingletonsInstantiated() {
```

Comment:
Let's stay away from `afterSingletonsInstantiated()` because it is not called from dynamic integration flows.

In other places we use `getDiscardChannel()` to resolve it lazily.

Reply:
I had a bunch of failing tests when it was in `onInit()` - I will take another look.

Reply:
No, that must not be in the `onInit()`. We need to introduce a `getDiscardChannel()`  with a similar lazy-init logic like we have in the `getOutputChannel()`